HopsanGUI/Utilities/GUIUtilities.cpp

```diff
@@ -374,6 +374,26 @@ QVector< complex<double> > realToComplex(const QVector<double> &rRealVector)
 }
 
 
+//! @brief Apply windowing function
+//! Only Hann windows are supported for now
+//! @param data Vector with data to be windowed
+void windowFunction(QVector<double> &data, WindowingFunctionEnumT function)
+{
+    switch(function) {
+        case RectangularWindow:
+        {
+            break;  //Rectangular shall do nothing
+        }
+        case HannWindow: {
+            int N = data.size()-1;
```

Comment:
should check that data is not empty

Reply:
Actually it does not matter. If data is empty N will become negative, and then the for loop below will not be executed.

Reply:
I realized that there will indeed by a division by zero here if data size is 1, so that N becomes 0. I added a check to prevent this.

Reply:
OK, I forgot that qt datatypes use int for size,